src/services/ipinfo-client.ts

```diff
@@ -0,0 +1,21 @@
+import axios from 'axios';
+
+export interface IpInfoResult {
+  data: { country: string };
+}
+
+export interface IpinfoClient {
+  getCountryCode(): Promise<IpInfoResult>;
+}
+
+export class HttpIpinfoClient implements IpinfoClient {
+  public async getCountryCode(): Promise<IpInfoResult> {
+    try {
+      const response = await axios.get('http://ipinfo.io/json');
```

Comment:
`axios` should be injected as a dependency here.

Reply:
I injected it at the top of the file, should I put it here instead of where I put it?